views/js/runtime/qtiCustomInteractionContext.js

```diff
@@ -75,31 +75,14 @@ define(function(){
          * @param {string} pciTypeIdentifier
          * @returns {Object} clonedPciModel
          */
-        createPciInstance : function(pciTypeIdentifier){
+        createPciInstance: function(pciTypeIdentifier) {
+            const registeredPCI = window._pciHooks[pciTypeIdentifier];
 
-            if(window._pciHooks[pciTypeIdentifier]){
-
-                var instance = {},
-                    proto = window._pciHooks[pciTypeIdentifier];
-
-                for(var name in proto){
-                    if(typeof proto[name] === 'function'){
-                        //@todo : delegate function call for better performance ?
-                        instance[name] = proto[name];
-                    }else if(proto[name] !== null && typeof proto[name] === 'object'){
-                        //a plain object:
-                        instance[name] = proto[name].constructor();
-                    }else{
-                        //not an object (nor a function) : e.g. 0, 123, '123', null, undefined
-                        instance[name] = proto[name];
-                    }
-                }
-
-                return instance;
-
-            }else{
+            if (!registeredPCI) {
                 throw new Error('no portable custom interaction hook found with the id ' + pciTypeIdentifier);
             }
+
+            return Object.assign({}, registeredPCI);
```

Comment:
The previous version was somehow doing a clone of nested objects. Should not it be the same here?

Reply:
Do you think  `instance[name] = proto[name].constructor();`  was creating a deep clone ?

Reply:
There are no problems with that.
The current method `createPciInstance` is factory for creating `registeredPCI` copies.
It simply provides copy of PCI interface implementation.
The issue with `instance[name] = proto[name].constructor();` it's overrides all PCI instance properties which has value `Object` type by empty object.

For instance, I have PCI:
```javascript
const customPCI = {
   init(){
        this.getState()
   },
   getState(){
       return this._state
   },
   _state: {
         value: 'Default state'
   }
}

const newPCI = createPciInstance(customPCI)

newPCI.init();
// Console output 
// without bugfix
// {}
// with bugfix
// { value: 'Default state'}
```

Reply:
I don't think `instance[name] = proto[name].constructor();` was creating a deep clone. However, it could have recreated the object if it was instanciated... (but without the`new` keyword I'm not so sure). Anyway, since it was just creating empty objects, I guess it's ok.

However, I'm still wondering if a simple assign will not introduce issue with object nested inside a PCI's definition. If 2 instances of the same PCI are living in the page, and each of them are mutating this object, what will happen?

Reply:
The possible issue which @jsconan mentioned above is not reproducible for liquids PCI since [the internal state assigned to the new value on update](https://github.com/oat-sa/extension-tao-itemqti-pci/blob/master/views/js/pciCreator/dev/liquidsInteraction/runtime/liquidInteraction.js#L291) (state is immutable object).
To prevent issue with mutability in the future, PCI creation is using `deepClone` for now.
Thanks @jsconan for good catch  👍